### PR TITLE
HDDS-6815: EC: getFileCheckSum should return null EC files until ECFileChecksum implemented.

### PR DESCRIPTION
**Pull request by umamaheswararao**
## What changes were proposed in this pull request?

GetFileChecksum is implemented only for Ratis based file layout. For EC files layout, we have no implementation ready yet. This JIRA is to return null for EC files until we have the implementation ready. Otherwise, we will get the wrong CRC unknowingly for EC files.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6815

## How was this patch tested?

Added a UT.

### CODE REVIEW

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneClientUtils.java [hunk 1]
```diff
@@ -205,8 +207,19 @@ public static FileChecksum getFileChecksumWithCombineMode(OzoneVolume volume,
...
     }
-    BaseFileChecksumHelper helper = new ReplicatedFileChecksumHelper(
-        volume, bucket, keyName, length, combineMode, rpcClient);
+    OmKeyArgs keyArgs = new OmKeyArgs.Builder().setVolumeName(volume.getName())
+        .setBucketName(bucket.getName()).setKeyName(keyName)
+        .setRefreshPipeline(true).setSortDatanodesInPipeline(true)
+        .setLatestVersionLocation(true).build();
+    OmKeyInfo keyInfo = rpcClient.getOzoneManagerClient().lookupKey(keyArgs);
+
+    if (keyInfo.getReplicationConfig()
```
Comment (jojochuang) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneClientUtils.java:216]: can you add a TODO comment here? Like:
```suggestion
    // TODO: return null util ECFileChecksum is implemented
    if (keyInfo.getReplicationConfig()
```

File: hadoop-ozone/ozonefs-common/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneClientUtils.java [hunk 2]
```diff
@@ -72,6 +76,34 @@ public void testEmptyKeyName() throws IOException {
...
+      Mockito.when(ozoneManagerProtocol.lookupKey(any())).thenReturn(omKeyInfo);
+      FileChecksum checksum = OzoneClientUtils
+          .getFileChecksumWithCombineMode(volume, bucket, "test", 1,
+              OzoneClientConfig.ChecksumCombineMode.MD5MD5CRC, clientProtocol);
+      assertNull(checksum);
+    } finally {
+      Mockito.reset(volume);
+      Mockito.reset(bucket);
+      Mockito.reset(volume);
+      Mockito.reset(volume);
```
Comment (jojochuang) [inline hadoop-ozone/ozonefs-common/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneClientUtils.java:103]: i'm not familiar with this. Is it required to reset volume multiple times?

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneClientUtils.java (hunk 1 above)
Comment (umamaheswararao) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneClientUtils.java:216]: Sure, I thought I added a comment somewhere. Let me do that right away.

File: hadoop-ozone/ozonefs-common/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneClientUtils.java (hunk 2 above)
Comment (umamaheswararao) [reply hadoop-ozone/ozonefs-common/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneClientUtils.java:103]: Oops, this is my mistake. I meant to clean all other mocks :-)

**Review by jojochuang — APPROVED** [review]

**Comment by jojochuang** [comment]
Merged. Thanks @umamaheswararao !

**Comment by umamaheswararao** [comment]
Thanks @jojochuang for the review!

**Pull request merged**